### PR TITLE
[react-sortable-tree] Fix TreeItem and ExtendedNodeData definition

### PR DESCRIPTION
**Pull request by dolezel**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/frontend-collective/react-sortable-tree
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/react-sortable-tree/index.d.ts [hunk 1]
```diff
@@ -20,8 +21,8 @@ export * from './utils/tree-data-utils';
 export * from './utils/default-handlers';
 
 export interface TreeItem {
-    title?: string;
-    subtitle?: string;
+    title?: React.ReactNode;
```
Comment (jzoric) [inline types/react-sortable-tree/index.d.ts]: Why not string?

File: types/react-sortable-tree/react-sortable-tree-tests.tsx [hunk 2]
```diff
@@ -68,9 +68,12 @@ class Test extends React.Component {
                     searchFinishCallback={(matches: NodeData[]) => {
                         const firstTitle = matches[0].node.title;
                     }}
-                    generateNodeProps={(data: ExtendedNodeData) => ({
-                        buttons: [data.node.title]
-                    })}
+                    generateNodeProps={(data: ExtendedNodeData) => {
+                        const { parentNode } = data;
```
Comment (jzoric) [inline types/react-sortable-tree/react-sortable-tree-tests.tsx]: Do we need the `parentNode` if it's not used?

**Review by jzoric — COMMENTED** [review]
Hi and thanks for the PR. I have two questions.

File: types/react-sortable-tree/index.d.ts (hunk 1 above)
Comment (dolezel) [reply types/react-sortable-tree/index.d.ts]: Title and subtitle can contain anything what can be rendered. (e.g. elements)

File: types/react-sortable-tree/react-sortable-tree-tests.tsx (hunk 2 above)
Comment (dolezel) [reply types/react-sortable-tree/react-sortable-tree-tests.tsx]: Didn't know how better update tests 🤷‍♂️

Comment (jzoric) [reply types/react-sortable-tree/react-sortable-tree-tests.tsx]: It would be OK without that line. We don't use other stuff either.

Comment (dolezel) [reply types/react-sortable-tree/react-sortable-tree-tests.tsx]: ok

**Review by wouterhardeman — APPROVED** [review]
This can be merged, thanks for contributing! The error is from another typing so it can be ignored.

**Review by jzoric — APPROVED** [review]

**Pull request merged**